sympy/parsing/sympy_parser.py

```diff
@@ -626,16 +620,91 @@ def convert_xor(tokens, local_dict, global_dict):
 
     return result
 
+def repeated_decimals(tokens, local_dict, global_dict):
+    """
+    Allows 0.2[1] notation to represent the repeated decimal 0.21... (19/90)
+
+    Run this before auto_number.
+
+    TODO: This would be much easier to implement as an AST transformer.
+    """
+    result = []
+
+    # num will running match any DECIMAL [ INTEGER ]
+    num = []
+    for toknum, tokval in tokens:
+        if toknum == NUMBER:
+            if (not num and '.' in tokval and 'e' not in tokval.lower() and
+                'j' not in tokval.lower() and 'x' not in tokval.lower()):
+                num.append((toknum, tokval))
+            elif tokval.isdigit() and len(num) == 2:
+                num.append((toknum, tokval))
+            else:
+                num = []
+        elif toknum == OP:
+            if tokval == '[' and len(num) == 1:
+                num.append((OP, tokval))
+            elif tokval == ']' and len(num) == 3:
+                num.append((OP, tokval))
+            elif tokval == '.' and not num:
+                # handle .[1]
+                num.append((NUMBER, '0.'))
+            else:
+                num = []
+        else:
+            num = []
+
+        result.append((toknum, tokval))
+        if len(num) == 4:
+            # pre.post[repeatend] = a + b/c + d/e where a = pre, b/c = post,
```

Comment:
repeatend -> repetend in this line and next

Reply:
Fixed, thanks.